src/compiler/checker.ts

```diff
@@ -1730,14 +1730,25 @@ namespace ts {
                 : resolveSymbol(moduleSymbol.exports.get(name), dontResolveAlias);
         }
 
+        function isSyntacticDefault(node: Node) {
+            if (isExportAssignment(node) && !node.isExportEquals) {
+                return true;
+            }
+            if (hasModifier(node, ModifierFlags.Default)) {
+                return true;
+            }
+            return false;

```

Comment:
Either one line, or two lines, but I think this is just as readable if you prefer
```ts
return isExportAssignment(node) && !node.isExportEquals
  || hasModifier(node, ModifierFlags.Default);
```

Reply:
+1 for @alfaproject suggestion.